demos/k8s-demo/rotate_password.sh

```diff
@@ -10,20 +10,22 @@ fi
 . ./utils.sh
 . ./admin_config.sh
 
-# update stored credentials
-update_password_k8s_secret "$new_password"
+# update db credentials + prune open connections
+docker run --rm -i postgres:9.6 env \
+ PGPASSWORD=$DB_ADMIN_PASSWORD psql -U $DB_ADMIN_USER "postgres://$DB_URL" -c "
+ALTER ROLE $DB_USER WITH PASSWORD '$new_password';
+"
 
-# wait for stored credentials to be propagated to application workloads
-while [[ ! "$(kubectl --namespace quick-start exec -it $(get_first_pod_for_app quick-start-application quick-start) -c secretless -- cat /etc/secret/password)" == "$new_password" ]] ; do
-    echo "Waiting for secret to be propagated"
-    sleep 10
-done
-echo Ready!
+# update stored credentials
+kubectl --namespace quick-start \
+ patch secret \
+ quick-start-backend-credentials \
+ -v=1 \
+ -p='{"data":{"password": "'$(echo -n "${new_password}" | base64)'"}}'
```

Comment:
I think it would be clearer if the json was defined as a variable before this kubectl command as this single line is incredibly logic-dense

Reply:
addressed